ariadne/cardano/test/backend/Test/Spec/AcidState.hs

```diff
@@ -0,0 +1,80 @@
+module Test.Spec.AcidState (spec) where
+
+import System.IO.Temp
+
+import Data.Acid (closeAcidState, openLocalStateFrom, query)
+import Data.Acid.Local (getState)
+
+import qualified Data.ByteString.Lazy as BL
+import Data.Text.Buildable (build)
+import Data.Text.Lazy.Builder (toLazyText)
+
+import Test.Hspec (Spec, describe)
+import Test.Hspec.QuickCheck (prop)
+import Test.QuickCheck (arbitrary, withMaxSuccess)
+import Test.QuickCheck.Monadic (monadicIO, pick, run)
+
+import Test.Spec.CreateWallet (NewWallet, applyNewWallet, genNewWalletRq)
+import Test.Spec.Fixture (genSpendingPasswords, withLayerLocalStorage)
+
+import Ariadne.Wallet.Cardano.Kernel.DB.AcidState (DB(..), Snapshot(..), defDB)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet (HdWallets(..))
+import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet (IxSet, size)
+import Ariadne.Wallet.Cardano.Kernel.Internal (PassiveWallet(..), wallets)
+import Ariadne.Wallet.Cardano.WalletLayer (PassiveWalletLayer, pwlCreateWallet)
+
+spec :: Spec
+spec = describe "Checking AcidState related functions behavior" $ do
+  prop "Check empty DB state" $ withMaxSuccess 1 $
+    monadicIO $ run $ withSystemTempDirectory "testWalletDBEmpty" $ \path -> do
+      getState path defDB 0 True >>= \case
+             Left err -> fail err
+             Right (acidDB, eventTag) -> do
+               db <- liftIO $ query acidDB Snapshot
+               let isEmptyDB = checkEmptyWalletDB db
+               return $ isEmptyDB && (eventTag == noEventsMsg)
+  prop "Check that opened state is the same as it was before closing." $ withMaxSuccess 10 $ do

```

Comment:
```suggestion
  prop "opened state is the same as it was before closing" $ withMaxSuccess 10 $ do
```

Reply:
I don't really understand this test. I reverted `stack.yaml` to the commit before your fixes and this test fails, but it fails somewhere inside acid-state rather than because of the checks you are doing here.

Reply:
In this test, I create <=10 wallets, save the list of that wallets and close the acid-state DB. 
Then I open it again, get a list of wallets and compare with the previously saved list. 
If there are any differences then the test fails.

Reply:
Added a handler for this exception.